src/sparsezoo/api/query_parser.py

```diff
@@ -93,12 +93,64 @@ def _parse_arguments(self) -> None:
 
     def _parse_fields(self) -> None:
         fields = self.fields or DEFAULT_FIELDS.get(self.operation_body)
-
+        if isinstance(fields, List):
+            self.fields = self.parse_list_fields_to_string(fields)
+        elif isinstance(fields, Dict):
+            """
+            fields = {
+                "model_id": None,
+                "benchmarks": {
+                    benchmark_result_id: None,
+                    whatever: {
+                        ...
+                    }
+                }
+            }
+            """
```

Comment:
What you have here is fine, don't feel like you need to change it, but I wanted to take the opportunity to point out an alternative as some food for thought. Some kinds of `if` statements can be replaced with a dictionary lookup. I think something like that could be possible here.
```python
FIELDS_PARSERS = {
    List: parse_list_fields_to_string,
    Dict: parse_dict_fields_to_string,
}
...

for fields_type, parser in FIELDS_PARSERS.items():
    if isinstance(fields, fields_type):
        self.fields = parser(fields)
        break
```
One benefit is that it makes the `_parse_fields` method better support the open/closed principle in that the addition of `FIELDS_PARSERS` makes it so new fields formats could be supported without having to change the logic in this method.

I defer to you on whether to take a similar approach. It's a bit awkward here since the parser functions are instance methods, if they could be made static methods or independent functions, that would help, but might come with its own set of tradeoffs.